datafusion/core/src/execution/context/mod.rs

```diff
@@ -468,6 +468,12 @@ impl SessionContext {
         self
     }
 
+    /// Register a [`CacheProducer`] to provide custom caching strategy
+    pub fn with_cache_producer(self, cache_producer: Arc<dyn CacheProducer>) -> Self {
```

Comment:
would it make sense to call it `cache_factory` rather than `cache_producer` to align with other similar methods like `function_factory`?

Reply:
do we need to expose this on session context, its not going to be used that often, maybe just having it on session state would be enough ?